### PR TITLE
Prepare release Hyperion v0.8.0

### LINKED ISSUE #238: Hyperion CHANGELOG
### Location
[CHANGELOG.md](https://github.com/hyperledger-labs/go-perun/blob/dev/CHANGELOG.md)

### Problem
Extend the changelog for *v0.8.0 Hyperion*.

### Proposal
Use something like `git log --merges v0.7.0..HEAD` and go through all merges and add them to the Changelog if needed.

### PR DESCRIPTION
**Pull request by matthiasgeihs**
Closes #238

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+## [0.8.0] Hyperion - 2021-11-08 [:warning:]
+Reorg-resistance for the Ethereum backend and support for external Watchtowers.
+
+### Added
+
+- :sparkles: **Ethereum backend: Reorg resistance** [#19](https://github.com/hyperledger-labs/go-perun/issues/19): The Ethereum backend now lets the user specify after how many blocks a transaction or an event should be considered confirmed.
+- :sparkles: **Watcher interface** [#172](https://github.com/hyperledger-labs/go-perun/issues/172): The watcher logic is now injectable. The adjudicator takes a watcher instance as a setup parameter. This enables using remote watcher services. A local watcher implementation is provided.
+
```
Comment (ggwpez) [inline CHANGELOG.md:16]: #240 Allocation helpers.

File: CHANGELOG.md [hunk 2]
```diff
@@ -255,3 +273,18 @@ Initial release.
...
+[#124]: https://github.com/hyperledger-labs/go-perun/pull/124
+[#144]: https://github.com/hyperledger-labs/go-perun/pull/144
+[#204]: https://github.com/hyperledger-labs/go-perun/pull/204
+[#129]: https://github.com/hyperledger-labs/go-perun/pull/129
+[#134]: https://github.com/hyperledger-labs/go-perun/pull/134
+[#148]: https://github.com/hyperledger-labs/go-perun/pull/148
+[#191]: https://github.com/hyperledger-labs/go-perun/pull/191
+[#196]: https://github.com/hyperledger-labs/go-perun/pull/196
+[#207]: https://github.com/hyperledger-labs/go-perun/pull/207
+[#218]: https://github.com/hyperledger-labs/go-perun/pull/218
```
Comment (sebastianst) [inline CHANGELOG.md:290]: `#123`'s are automatically translated by Github, no need for this section (which would grow unnecessarily long if we followed this pattern in the future). You can just remove the `[...]` around them in above text. See e.g. [geth release notes](https://github.com/ethereum/go-ethereum/releases/tag/v1.10.12), they also just mention the issue/PR numbers as `#123`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+Reorg-resistance for the Ethereum backend and support for external Watchtowers.
+
+### Added
+
+- :sparkles: **Ethereum backend: Reorg resistance** [#19](https://github.com/hyperledger-labs/go-perun/issues/19): The Ethereum backend now lets the user specify after how many blocks a transaction or an event should be considered confirmed.
+- :sparkles: **Watcher interface** [#172](https://github.com/hyperledger-labs/go-perun/issues/172): The watcher logic is now injectable. The adjudicator takes a watcher instance as a setup parameter. This enables using remote watcher services. A local watcher implementation is provided.
+
+### Changed
+
+- :arrow_up: Update go to v1.17 and go-ethereum to v1.10.8.
```
Comment (sebastianst) [inline CHANGELOG.md:19]: We usually take a release as an opportunity to update geth to the latest version (and maybe the other dependencies, too). Maybe update to [geth 1.10.12](https://github.com/ethereum/go-ethereum/releases/tag/v1.10.12) (and see if the tests still run).

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.8.0] Hyperion - 2021-11-08 [:warning:]
+Reorg-resistance for the Ethereum backend and support for external Watchtowers.
+
+### Added
+
+- :sparkles: **Ethereum backend: Reorg resistance** [#19](https://github.com/hyperledger-labs/go-perun/issues/19): The Ethereum backend now lets the user specify after how many blocks a transaction or an event should be considered confirmed.
+- :sparkles: **Watcher interface** [#172](https://github.com/hyperledger-labs/go-perun/issues/172): The watcher logic is now injectable. The adjudicator takes a watcher instance as a setup parameter. This enables using remote watcher services. A local watcher implementation is provided.
```
Comment (sebastianst) [inline CHANGELOG.md:15]:
```suggestion
- :sparkles: **Ethereum backend: Reorg resistance** (#19): The Ethereum backend now lets the user specify after how many blocks a transaction or an event should be considered confirmed.
- :sparkles: **Watcher interface** (#172): The watcher logic is now injectable. The adjudicator takes a watcher instance as a setup parameter. This enables using remote watcher services. A local watcher implementation is provided.
```
Github automatically translates issue and PR numbers into the correct links.

File: CHANGELOG.md (hunk 2 above)
Comment (matthiasgeihs) [reply CHANGELOG.md:290]: This is not the case, see [here](https://github.com/perun-network/go-perun/blob/f57bff55dcf119a85cd0f5766ebf681b8078240d/CHANGELOG.md).

File: CHANGELOG.md (hunk 3 above)
Comment (matthiasgeihs) [reply CHANGELOG.md:19]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (matthiasgeihs) [reply CHANGELOG.md:15]: In some locations GitHub does this, but apparently not when viewing the [CHANGELOG.md](https://github.com/perun-network/go-perun/blob/f57bff55dcf119a85cd0f5766ebf681b8078240d/CHANGELOG.md).

File: CHANGELOG.md (hunk 2 above)
Comment (sebastianst) [reply CHANGELOG.md:290]: Sorry, you're totally right. Don't know why I thought that...
I still propose that we inline those links and not put them at the end of the document as footnotes.

File: CHANGELOG.md (hunk 4 above)
Comment (sebastianst) [reply CHANGELOG.md:15]: It doesn't 🙈

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Changed
+
+- :arrow_up: Update go to v1.17 and go-ethereum to v1.10.8.
+- :white_check_mark: Extend support for additional blockchain backends by revising the generic tests (e.g., [#225], [#227], [#228]).
+
+### Fixed
+- :children_crossing: Improve usability (e.g., [#124], [#144], [#204]).
```
[the remark this answers is not in the document]
Comment (matthiasgeihs) [reply CHANGELOG.md:23]: What do you mean by "the old links"?

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,23 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- :bug: Improve stability (e.g., [#129], [#134], [#148], [#191], [#196], [#207], [#218]).
```
[the remark this answers is not in the document]
Comment (matthiasgeihs) [reply CHANGELOG.md:24]: Moved to "Changed"

File: CHANGELOG.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ggwpez) [reply CHANGELOG.md:23]: Okay they dont, fail.

File: CHANGELOG.md (hunk 2 above)
Comment (matthiasgeihs) [reply CHANGELOG.md:290]: I definitely agree with your point that this is somehow polluting the CHANGELOG. Open for suggestions. One alternative of course is to not link the issues.

Comment (sebastianst) [reply CHANGELOG.md:290]: Hmm I thought that this is somehow visible in the final CHANGELOG, but it isn't. Maybe we move this list of links always at the end of each changelog entry (so above the title of the next changelog entry)? Then it doesn't grow into a very long list and stays together with the entry.

**Review by sebastianst — COMMENTED** [review]
See one open comment.

**Comment by matthiasgeihs** [comment]
Squashed. Fixed the go-ethereum version in the CHANGELOG to 1.10.12.

**Review by sebastianst — APPROVED** [review]

**Pull request merged**